its/it/azdo.ts

```diff
@@ -9,6 +22,46 @@ export function getAzdoApi(): vm.WebApi {
   return new vm.WebApi(AZDO_BASE_URL + AZDO_ORGANIZATION, credentialHandler);
 }
 
+export async function getTemplatePipeline(buildApi: IBuildApi): Promise<BuildDefinition> {
```

Comment:
I create the new pipelines based on the existing ones because if not, we need some additional logic to create or find a GitHub service connection ID to register and link when creating the new pipeline, and we need to have some additional logic to get the repository information (ID, URL, etc)

Reply:
https://dev.azure.com/sonarsource/SonarScannerAzdo/_releaseProgress?_a=release-environment-logs&releaseId=11&environmentId=11

I am confused by the logs here

Why is this one pipeline used as a template, its the first returned?
`2024-11-07T13:22:52.8337798Z Using pipeline pipeline-sonarcloud-v3-cli-embedded-unix as a template`

And the additional logs, I can't see why some are skipped and others not (were you testing the creation scenario? I thought they were all existing already 🤷🏻 )
```
2024-11-07T13:22:52.8337798Z Using pipeline pipeline-sonarcloud-v3-cli-embedded-unix as a template
2024-11-07T13:22:52.8907300Z Pipeline pipeline-sonarcloud-v1-cli-6.2.1.4610-unix already exists. Skipping creation.
2024-11-07T13:22:52.9940286Z Pipeline pipeline-sonarcloud-v1-cli-6.2.1.4610-windows already exists. Skipping creation.
2024-11-07T13:22:53.0617157Z Pipeline pipeline-sonarcloud-v1-cli-embedded-unix already exists. Skipping creation.
2024-11-07T13:22:53.1153584Z Pipeline pipeline-sonarcloud-v1-cli-embedded-windows already exists. Skipping creation.
2024-11-07T13:22:53.1868942Z Pipeline pipeline-sonarcloud-v1-dotnet-8.0.3.99785-unix already exists. Skipping creation.
2024-11-07T13:22:53.2358409Z Pipeline pipeline-sonarcloud-v1-dotnet-8.0.3.99785-windows already exists. Skipping creation.
2024-11-07T13:22:53.2894280Z Pipeline pipeline-sonarcloud-v1-dotnet-embedded-unix already exists. Skipping creation.
2024-11-07T13:22:53.3559588Z Pipeline pipeline-sonarcloud-v1-dotnet-embedded-windows already exists. Skipping creation.
2024-11-07T13:22:53.5033930Z Creating pipeline pipeline-sonarcloud-v1-other-gradle-unix
2024-11-07T13:22:56.3261009Z Creating pipeline pipeline-sonarcloud-v1-other-gradle-windows
2024-11-07T13:22:59.4497797Z Creating pipeline pipeline-sonarcloud-v1-other-maven-unix
2024-11-07T13:23:02.3549252Z Creating pipeline pipeline-sonarcloud-v1-other-maven-windows
2024-11-07T13:23:05.7562444Z Creating pipeline pipeline-sonarcloud-v2-cli-6.2.1.4610-unix
2024-11-07T13:23:09.1293860Z Creating pipeline pipeline-sonarcloud-v2-cli-6.2.1.4610-windows
2024-11-07T13:23:12.1795917Z Creating pipeline pipeline-sonarcloud-v2-cli-embedded-unix
2024-11-07T13:23:15.5702818Z Creating pipeline pipeline-sonarcloud-v2-cli-embedded-windows
2024-11-07T13:23:21.0827246Z Creating pipeline pipeline-sonarcloud-v2-dotnet-8.0.3.99785-unix
2024-11-07T13:23:24.3738178Z Creating pipeline pipeline-sonarcloud-v2-dotnet-8.0.3.99785-windows
2024-11-07T13:23:27.8209562Z Creating pipeline pipeline-sonarcloud-v2-dotnet-embedded-unix
2024-11-07T13:23:30.9816788Z Pipeline pipeline-sonarcloud-v2-dotnet-embedded-windows already exists. Skipping creation.
2024-11-07T13:23:31.0569229Z Creating pipeline pipeline-sonarcloud-v2-other-gradle-unix
2024-11-07T13:23:34.9554137Z Creating pipeline pipeline-sonarcloud-v2-other-gradle-windows
2024-11-07T13:23:38.1403197Z Creating pipeline pipeline-sonarcloud-v2-other-maven-unix
2024-11-07T13:23:41.2849781Z Creating pipeline pipeline-sonarcloud-v2-other-maven-windows
2024-11-07T13:23:44.5094015Z Creating pipeline pipeline-sonarcloud-v3-cli-6.2.1.4610-unix
2024-11-07T13:23:47.8413240Z Pipeline pipeline-sonarcloud-v3-cli-6.2.1.4610-windows already exists. Skipping creation.
2024-11-07T13:23:48.0102571Z Pipeline pipeline-sonarcloud-v3-cli-embedded-unix already exists. Skipping creation.
2024-11-07T13:23:48.0587292Z Creating pipeline pipeline-sonarcloud-v3-cli-embedded-windows
2024-11-07T13:23:51.2617180Z Pipeline pipeline-sonarcloud-v3-dotnet-8.0.3.99785-unix already exists. Skipping creation.
2024-11-07T13:23:51.3186232Z Pipeline pipeline-sonarcloud-v3-dotnet-8.0.3.99785-windows already exists. Skipping creation.
2024-11-07T13:23:51.3873479Z Pipeline pipeline-sonarcloud-v3-dotnet-embedded-unix already exists. Skipping creation.
2024-11-07T13:23:51.4406842Z Pipeline pipeline-sonarcloud-v3-dotnet-embedded-windows already exists. Skipping creation.
2024-11-07T13:23:51.5095313Z Pipeline pipeline-sonarcloud-v3-other-gradle-unix already exists. Skipping creation.
2024-11-07T13:23:51.5563685Z Creating pipeline pipeline-sonarcloud-v3-other-gradle-windows
2024-11-07T13:23:54.7784087Z Pipeline pipeline-sonarcloud-v3-other-maven-unix already exists. Skipping creation.
2024-11-07T13:23:54.8446187Z Creating pipeline pipeline-sonarcloud-v3-other-maven-windows
```

Reply:
> Why is this one pipeline used as a template, its the first returned?

yes

> I thought they were all existing already 🤷🏻

The ones that wern't created are the ones that we created before (that you are already using in your pull request)